squad/frontend/tests.py

```diff
@@ -170,13 +174,19 @@ def tests(request, group_slug, project_slug, build_version):
 
 
 @auth
-def test_history(request, group_slug, project_slug, build_version, testrun, suite_slug, test_name):
+def test_history(request, group_slug, project_slug, build_version=None, testrun=None, suite_slug=None, test_name=None):
     project = request.project
-    build = get_build(project, build_version)
-    test_run = get_object_or_404(build.test_runs, pk=testrun)
-    suite_slug = suite_slug.replace('$', '/')
-    suite = get_object_or_404(project.suites, slug=suite_slug)
-    status = get_object_or_404(test_run.status, suite=suite)
+    context = {"project": project}
+    if build_version and testrun and suite_slug:
+        build = get_build(project, build_version)
+        test_run = get_object_or_404(build.test_runs, pk=testrun)
+        suite_slug = suite_slug.replace('$', '/')
+        suite = get_object_or_404(project.suites, slug=suite_slug)
+        status = get_object_or_404(test_run.status, suite=suite)
+        full_test_name = "/".join([suite_slug, test_name])
+        context.update({"build": build, "status": status, "test": test_name})
+    else:
+        full_test_name = request.path.rsplit("tests/", 1)[1].replace('$', '/')
```

Comment:
this is risky. If suite or test contains 'tests/' in the name it will result in incorrect name.